### PR TITLE
Add on node processing

### PR DESCRIPTION
**Pull request by TApplencourt**
This adds some  `on-node` processing. 

Need to add some more options to add flexibility on where to save data and with step / processes to perform.

(Sorry for some typo change, it's rubocopt doing is thing...) 

- Added a `no-muxer` to babeltrace_thapi
- Add this `on-node-processing` concept to xprof.

### CODE REVIEW

File: xprof/xprof.rb.in [hunk 1]
```diff
@@ -370,132 +420,151 @@ def enable_events_cl(lttng_session_uuid, channel_name, tracing_mode: 'default',
...
   backends = OPTIONS[:'backend-name'].join(',')
   backend_level = OPTIONS[:backend].filter { |nl| nl.include?(':') }.join(',')
 
   if OPTIONS.include?(:trace)
-    opts = 'trace'
+    if folder.include?('_interval') || folder.include?('_aggreg')
+      raise 'Cannot show the trace of an interval or aggregated trace'
+    end
+
+    opts = 'trace '
```
Comment (TApplencourt) [inline xprof/xprof.rb.in:560]: This fix `trace` not working...  :( So we should merge ASAP, or can do a smaller PR with only those `space` fix.

File: xprof/xprof.rb.in [hunk 2]
```diff
@@ -370,145 +420,166 @@ def enable_events_cl(lttng_session_uuid, channel_name, tracing_mode: 'default',
...
-    opts << '--display_kernel_verbose true ' if OPTIONS.include?(:'kernel-verbose')
-    opts << '--display_metadata true ' if OPTIONS.include?(:metadata)
-    opts << "--display_name_max_size #{OPTIONS[:'max-name-size']} "
-    opts << '--display_mode json ' if OPTIONS.include?(:json)
-    opts << "--backend #{backends} "
+    opts = ['tally']
+    opts << '--display_kernel_verbose true' if OPTIONS.include?(:'kernel-verbose')
+    opts << '--display_metadata true' if OPTIONS.include?(:metadata)
+    opts << "--display_name_max_size #{OPTIONS[:'max-name-size']}"
+    opts << '--display_mode json' if OPTIONS.include?(:json)
```
Comment (Kerilk) [inline xprof/xprof.rb.in:574]:
```suggestion
    opts << '--display_mode' << 'json' if OPTIONS.include?(:json)
```
Wouldn't that prevent issues like space escaping if we end up using the Array in an exec call?

Comment (TApplencourt) [reply xprof/xprof.rb.in:574]: So weird issue with `-` and `_` in the variable. I got some error `incorrect --display-name-max-size did you mean display_name_max_size`

Comment (TApplencourt) [reply xprof/xprof.rb.in:574]: If we do it for `json` do you want to do it for all the other? like `max-name-size`  `true` and other?

Comment (TApplencourt) [reply xprof/xprof.rb.in:574]: Yes it was that
```
    opts = ['tally']
    opts << '--display_kernel_verbose' if OPTIONS.include?(:'kernel-verbose')
    opts << '--display_metadata' if OPTIONS.include?(:metadata)
    opts << "--display_name_max_size" << OPTIONS[:'max-name-size'].to_s
    opts << '--display_mode' << 'json' if OPTIONS.include?(:json)
    opts << "--backend" << backends
    opts << "--backend_level" << backend_level unless backend_level.empty?
    opts << '--display' << 'extended' if OPTIONS.include?(:extended)
  end

  opts << '--no-muxer' if folder.include?('thapi_interval') || folder.include?('thapi_aggreg')

  cmd = ["#{BINDIR}/babeltrace_thapi"] + opts + ["--", folder]
  ```

Is working

File: xprof/xprof.rb.in [hunk 3]
```diff
@@ -370,147 +420,170 @@ def enable_events_cl(lttng_session_uuid, channel_name, tracing_mode: 'default',
...
+    if folder.include?('_interval') || folder.include?('_aggreg')
+      raise 'Cannot show the trace of an interval or aggregated trace'
+    end
+
+    args += ['trace', '--restrict', '--context', '--backend', backends]
   elsif OPTIONS.include?(:timeline)
-    opts = 'timeline'
-    opts << "--backend #{backends} "
+    raise 'Cannot show the trace of an aggreg trace' if folder.include?('_aggreg')
+    args += ['timeline' '--backend', backends]
```
Comment (Kerilk) [inline xprof/xprof.rb.in:568]:
```suggestion
    args += ['timeline', '--backend', backends]
```

**Review by Kerilk — APPROVED** [review]

**Pull request merged**